<!DOCTYPE html>
<html>

<style>
seq {font-family:Monaco, "Courier New";
  font-size:0.9em;
  text-indent: 2.7em;
}

align.small {
    font-family:Monaco, "Courier New";
    font-size:0.9em;
    line-height:65%;
}

align.long {
  font-family:Monaco, "Courier New";
  font-size:0.8em;
  line-height:70%;
}

s5 {color:#6baed6;}
s7 {color:#fc9272;}
p5 {color:#08519c;}
p7 {color:#a50f15;}
me {color:#969696;}
tso {color:#2ca25f;}
cbc {color:#f768a1;}
umi {color:#807dba;}
t7 {color:blue;}
pe1 {color:#bcbddc;}
pe2 {color:#9ebcda;}
w1 {color:#f03b20;}
h3 {font-family:verdana;}

</style>

<head>
<title>inDrop</title>
</head>
<body>

<h1><a href="http://www.cell.com/cell/abstract/S0092-8674(15)00500-0" target="_blank">inDrop</a></h1>

<p><span style="font-family:verdana; font-size:1.1em;">inDrop was originally pulished in Cell 161, 1187–1201 (2015). Then, two year after that, the authors published the detailed protocol in Nature Protocols 12, 44–73 (2017), which has different primer design comparing to the orginal paper. The workflow shown here is based on the protocol in Nature Protocols 12, 44–73 (2017), which seems to be the version 2 of the inDrop technology. According to the <a href="https://github.com/indrops/indrops" target="_blank">inDrop github page</a>, there is a version 3, but the oligos and library structures are exactly the same as version 2, except the sequencing mode changed. In version 2, three different reads are generated; in vernsion 3, four different reads are generated. See details below.</span></p>

<br></br>

<h2>Adapter and primer sequences:</h2>
<h3>Sequence used during the experiment:</h3>
<seq>
<p><sup>*</sup>Beads-oligo-dT19V: |--5'- /5Acryd/iSpPC/CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>AAGGCGTCACAAGCAATCACTC</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>TTTTTTTTTTTTTTTTTTV -3'</p>
<p>T7 promoter: 5'- <t7>TAATACGACTCACTATAGGG</t7> -3'</p>
<p>PE1 adapter: 5'- <pe1>CTCTTTCCCTACACGACGCTCTTC</pe1> -3'</p>
<p>PE2 adapter: 5'- <pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2> -3'</p>
<p>PE2-N6 RT primer: 5'- <pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>NNNNNN -3'</p>
<p>W1 sequence: 5'- <w1>AAGGCGTCACAAGCAATCACTC</w1> -3'</p>
<p>PE1 PCR primer: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[6-bp sample index]<pe1>CTCTTTCCCTACACGA</pe1> -3'</p>
<p>PE2 PCR primer: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>GGTC<pe2>TCGGCATTCCTGCTGAAC</pe2> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'</p>
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'</p>
<p>Read 1 sequencing primer: 5'- <pe2>GGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2> -3'</p>
<p>Index sequencing primer: 5'- AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1> -3'</p>
<p>Read 2 sequencing primer: 5'- <pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT -3'</p>
<p><b><span style="color:red">*</span></b> barcode1 has variable length.</p>
</seq>
<h3>Sequence used for barcoded oligo-dT19V generation (before the experiment):</h3>
<seq>
<p>Acrydite-modified primer: 5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1> -3'</p>
<p>BA19 oligo: 5'- BAAAAAAAAAAAAAAAAAAA -3'</p>
<p>FAM-PE1 probe: 5'-/6-FAM/ AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1> -3'</p>
<p>FAM-W1 probe: 5'-/6-FAM/ <w1>AAGGCGTCACAAGCAATCACTC</w1> -3'</p>
<p>FAM-BA19 probe: 5'-/6-FAM/ BAAAAAAAAAAAAAAAAAAA -3'</p>
<p>Barcode 1 plates (384 of them in 384 individual wells): 5'- <w1>AAGGCGTCACAAGCAATCACTC</w1><cbc>[barcode1]</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1> -3', for the sequences of 384 different barcode1, see <a href="data/inDrop_barcode1_list.txt" target="_blank">inDrop_barcode1_list.txt</a>.</p>
<p>Barcode 2 plates (384 of them in 384 individual wells): 5'- BAAAAAAAAAAAAAAAAAAA<umi>[6-bp UMI]</umi><cbc>[8-bp barcode2]</cbc><w1>AAGGCGTCACAAGCAATCACTC</w1> -3', for the sequences of 384 different barcode2 (8 bp), see <a href="data/inDrop_barcode2_list.txt" target="_blank">inDrop_barcode2_list.txt</a>.</p>
</seq>

<br></br>

<h2>Step-by-step generation of oligo-dT19V:</h2>
<h3>(1) Bind Acrydite-modified primer to acrylimide gel beads by mixing:</h3>
<pre>
<seq>
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1> -3'
</seq>
</pre>

<h3>(2) Split the gel beads with bound primers into wells in Barcode 1 plates, and primer extension:</h3>
<pre>
<seq>
                                                      <--------<pe1>GAGAAAGGGATGTGCTGCGAGAAG</pe1>GCTAGA<cbc>[barcode1]</cbc><w1>CTCACTAACGAACACTGCGGAA</w1> -5'
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>-------->
</seq>
</pre>

<h3>(3) This is the product after the first extension:</h3>
<pre>
<seq>
                   5'- GCTACTGC<t7>ATTATGCTGAGTGATATCCC</t7>TATGGTGGTACC<pe1>GAGAAAGGGATGTGCTGCGAGAAG</pe1>GCTAGA<cbc>[barcode1]</cbc><w1>CTCACTAACGAACACTGCGGAA</w1> -5'
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1>
</seq>
</pre>

<h3>(4) Pooling, denature by NaOH, and get rid of top strand:</h3>
<pre>
<seq>
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1>
</seq>
</pre>

<h3>(5) Split again into wells in Barcode 2 plates, and primer extension:</h3>
<pre>
<align class="long">
                                                                                              <--------<w1>CTCACTAACGAACACTGCGGAA</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>A<sub>19</sub>B -5'
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1>---------->
</align>
</pre>

<h3>(6) This is the product of second extension:</h3>
<pre>
<align class="long">
                   5'- GCTACTGC<t7>ATTATGCTGAGTGATATCCC</t7>TATGGTGGTACC<pe1>GAGAAAGGGATGTGCTGCGAGAAG</pe1>GCTAGA<cbc>[barcode1]</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>A<sub>19</sub>B -5'
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>T<sub>19</sub>V -3'
</align>
</pre>

<h3>(7) Pool aggain, denautre by NaOH, get rid of top strand, neutralise, wash and ready to use (384 x 384 different combination of barcodes 1 & 2):</h3>
<pre>
<align class="long">
|--5'-/acrydite/iSpPC/ CGATGACG<t7>TAATACGACTCACTATAGGG</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>T<sub>19</sub>V -3'
</align>
</pre>

<br></br>

<h2>Step-by-step library generation (the 5'-/acrydite/iSpPC/ is omitted for simplicity)</h2>
<h3>(1) Anneal oligo-dT19V to mRNA and reverse transcription using MMLV inside droplets:</h3>
<pre>
<align class="long">
5'- XXXXXXXXXXXXXXXXXXXB(A)<sub>n</sub>
                 <-----V(T)<sub>19</sub><umi>[6-bp UMI]</umi><cbc>[8-bp barcode2]</cbc><w1>AAGGCGTCACAAGCAATCACTC</w1><cbc>[barcode1]</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>GGTACCACCATA<t7>GGGATATCACTCAGCATAAT</t7>GCAGTAGC -5'--|
</align>
</pre>

<h3>(2) RNaseH and DNA Pol I based second strand synthesis:</h3>
<pre>
<align class="long">
5'- XXX...XXXB(pA)<umi>[6-bp UMI]</umi><cbc>[8-bp barcode2]</cbc><w1>TTCCGCAGTGTTCGTTAGTGAG</w1><cbc>[barcode1]</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>CCATGGTGGTAT<t7>CCCTATAGTGAGTCGTATTA</t7>CGTCATCG
    XXX...XXXV(dT)<umi>[6-bp UMI]</umi><cbc>[8-bp barcode2]</cbc><w1>AAGGCGTCACAAGCAATCACTC</w1><cbc>[barcode1]</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>GGTACCACCATA<t7>GGGATATCACTCAGCATAAT</t7>GCAGTAGC -5'--|
</align>
</pre>

<h3>(3) Break droplets and T7 <i>in vitro</i> transcription to amplify cDNA (resulting in single stranded RNA):</h3>
<pre>
<seq>
5'- <t7>G</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>AAGGCGTCACAAGCAATCACTC</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>(dT)VXXX...XXX -3'
</seq>
</pre>

<h3>(4) Reverse transcription on amplified RNA (aRNA) using random priming (PE2-N6)</h3>
<pre>
<align class="long">
5'- <t7>G</t7>ATACCACCATGG<pe1>CTCTTTCCCTACACGACGCTCTTC</pe1>CGATCT<cbc>[barcode1]</cbc><w1>AAGGCGTCACAAGCAATCACTC</w1><cbc>[8-bp barcode2]</cbc><umi>[6-bp UMI]</umi>(dT)VXXX...XXX -3'
                                                                                                      <-------NNNNNN
                                                                                                                    <pe2>TCTAGCCTTCTCGCCAAGTCGTCCTTACGGCT</pe2> -5'
</align>
</pre>

<h3>(5) Fist strand cDNA of aRNA looks like this:</h3>
<pre>
<align class="long">
5'- <pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>XXX...XXXV(pA)<umi>[6-bp UMI]</umi><cbc>[8-bp barcode2]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>[barcode1]</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>CCATGGTGGTAT<t7>C</t7> -3'
</align>
</pre>

<h3>(6) Adding PE1 PCR and PE2 PCR primers to amplify the first strand cDNA:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>GGTC<pe2>TCGGCATTCCTGCTGAAC</pe2>---------->
                                 5'- <pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>XXX...XXXV(pA)<umi>[6-bp UMI]</umi><cbc>[8-bp barcode2]</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>[barcode1]</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>CCATGGTGGTAT<t7>C</t7> -3'
                                                                                                                                               <----------<pe1>AGCACATCCCTTTCTC</pe1>
                                                                                                                                                                          [6-bp sample index]<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(9) Final library structure:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>GGTC<pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>NN....NN</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>NNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7>
    <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>CCAG<pe2>AGCCGTAAGGACGACTTGGCGAGAAGGCTAGA</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>NN....NN</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>NNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
            <p5>Illumina P5</p5>                           <pe2>PE2</pe2>                   cDNA        <umi>6bp</umi>    <cbc>8bp</cbc>            <w1>W1</w1>           <cbc>barcode1</cbc>               <pe1>PE1</pe1>             6bp         <p7>Illumina P7</p7>
                                                                                    <umi>UMI</umi>  <cbc>barcode2</cbc>                                                          sample index
</align>
</pre>

<br></br>

<h2>V2 Library sequencing (three steps):</h2>

<h3>(1) Add read 1 sequencing primer to sequence the first read (bottom strand as template, these are the cDNA reads):</h3>
<pre>
<align class="long">
                                   5'- <pe2>GGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>--------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>CCAG<pe2>AGCCGTAAGGACGACTTGGCGAGAAGGCTAGA</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>NN....NN</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>NNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add Index sequencing primer to sequence sample index (bottom strand as template):</h3>
<pre>
<align class="long">
                                                                                                                           5'- AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>----->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>CCAG<pe2>AGCCGTAAGGACGACTTGGCGAGAAGGCTAGA</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>NN....NN</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>NNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Cluster regeneration, and add read 2 sequencing primer to sequence read 2 (top strand as template, these are the cell barcodes and UMI reads, at least 51 cycles):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>GGTC<pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>NN....NN</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>NNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                                   <-------------------------------------------TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1> -5'
</align>
</pre>

<br></br>

<h2>V3 Library sequencing (four steps), this is just a guess based on the github page description and sample fastq files, not confirmed:</h2>

<h3>(1) Add read 1 sequencing primer to sequence the first read (bottom strand as template, these are the cDNA reads):</h3>
<pre>
<align class="long">
                                   5'- <pe2>GGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>--------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>CCAG<pe2>AGCCGTAAGGACGACTTGGCGAGAAGGCTAGA</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>NN....NN</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>NNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add reverse complement of W1 primer primer to sequence barcode1 (bottom strand as template, 6 cycles):</h3>
<pre>
<align class="long">
                                                                                             5'- <w1>GAGTGATTGCTTGTGACGCCTT</w1>----->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>CCAG<pe2>AGCCGTAAGGACGACTTGGCGAGAAGGCTAGA</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>NN....NN</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>NNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Add Index sequencing primer to sequence sample index (bottom strand as template, 6 cycles)::</h3>
<pre>
<align class="long">
                                                                                                                           5'- AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>----->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>CCAG<pe2>AGCCGTAAGGACGACTTGGCGAGAAGGCTAGA</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>CTCACTAACGAACACTGCGGAA</w1><cbc>NN....NN</cbc>TCTAGC<pe1>CTTCTCGCAGCACATCCCTTTCTC</pe1>NNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(4) Cluster regeneration, and add W1 primer to sequence barcode2 and UMI (top strand as template, seems to be 21 cycles, contain a bit of dT at the end):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>GGTC<pe2>TCGGCATTCCTGCTGAACCGCTCTTCCGATCT</pe2>XXX...XXXV(pA)<umi>NNNNNN</umi><cbc>NNNNNNNN</cbc><w1>GAGTGATTGCTTGTGACGCCTT</w1><cbc>NN....NN</cbc>AGATCG<pe1>GAAGAGCGTCGTGTAGGGAAAGAG</pe1>NNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                            <--------------------<w1>CTCACTAACGAACACTGCGGAA</w1> -5'
</align>
</pre>

</body>
</html>
