<!DOCTYPE html>
<html>

<style>
seq {font-family:Monaco, "Courier New";
	font-size:0.9em;
	text-indent: 20px;
}

align.small {
    font-family:Monaco, "Courier New";
    font-size:0.7em;
    line-height:65%;
}

align.big {
	font-family:Monaco, "Courier New";
	font-size:1em;
	line-height:100%;
}

s5 {color:#6baed6;}
s7 {color:#fc9272;}
p5 {color:#08519c;}
p7 {color:#a50f15;}
me {color:#969696;}
t7 {color:blue;}
umi {color:#807dba;}
cbc {color:#f768a1;}
tso {color:#2ca25f;}
h3 {font-family:verdana;}

</style>

<head>
<title>plate_scATAC-seq and Pi-ATAC-seq</title>
</head>
<body>

<h1><a href="https://www.nature.com/articles/s41467-018-07771-0" target="_blank">plate_scATAC-seq</a>
/ <a href="https://www.nature.com/articles/s41467-018-07115-y" target="_blank">Pi-ATAC-seq</a></h1>

<p><span style="font-family:verdana; font-size:1.1em;">It has been previously demonstrated that Tn5 transposase-mediated tagmentation contains two stages: (1) a tagging stage where the Tn5 transposome binds to DNA, and (2) a fragmentation stage where the Tn5 transposase is released from DNA using heat or denaturing agents, such as sodium dodecyl sulfate (SDS). As the Tn5 tagging does not fragment DNA, the nuclei would remain intact after incubation with the Tn5 transposome in an bulk ATAC-seq experiment. Both plate_scATAC-seq and Pi-ATAC-seq utilise this property of Tn5 tagmentation, and perform bulk tagging upfront, followed by FACS sorting of single nuclei into wells for library construction. FACS index sorting can also be reported. These methods are simple to follow and work robustly. Since each cells are processed seprately, the library indices are single cell barcodes.</span></p>

<br></br>

<h2>Adapter and primer sequences:</h2>
<seq>
<p>Nextera Tn5 binding site (19-bp Mosaic End (ME)): 5'- <me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Nextera N/S5xx primer entry point (s5): 5'- <s5>TCGTCGGCAGCGTC</s5> -3'</p>
<p>Nextera N7xx primer entry point (s7): 5'- <s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'
<p>Nextera (XT) N/S5xx Index primer: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[8-bp i5 index]<s5>TCGTCGGCAGCGTC</s5> -3'</p>
<p>Nextera (XT) N7xx Index primer: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[8-bp i7 index]<s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>Read 1 sequencing primer: 5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Index 1 sequencing primer: 5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7> -3'</p>
<p>Read 2 sequencing primer: 5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me> -3'</p>
</seq>

<h3>8-bp i5 & i7 sequences:</h3>
<pre>
    N/S502 : CTCTCTAT
    N/S503 : TATCCTCT
    N/S505 : GTAAGGAG
    N/S506 : ACTGCATA
    N/S507 : AAGGAGTA
    N/S508 : CTAAGCCT
    N/S510 : CGTCTAAT
    N/S511 : TCTCTCCG
    N/S513 : TCGACTAG
    N/S515 : TTCTAGCT
    N/S516 : CCTAGAGT
    N/S517 : GCGTAAGA
    N/S518 : CTATTAAG
    N/S520 : AAGGCTAT
    N/S521 : GAGCCTTA
    N/S522 : TTATGCGA

    N701 : TCGCCTTA
    N702 : CTAGTACG
    N703 : TTCTGCCT
    N704 : GCTCAGGA
    N705 : AGGAGTCC
    N706 : CATGCCTA
    N707 : GTAGAGAG
    N710 : CAGCCTCG
    N711 : TGCCTCTT
    N712 : TCCTCTAC
    N714 : TCATGAGC
    N715 : CCTGAGAT
    N716 : TAGCGAGT
    N718 : GTAGCTCC
    N719 : TACTACGC
    N720 : AGGCTCCG
    N721 : GCAGCGTA
    N722 : CTGCGCAT
    N723 : GAGCGCTA
    N724 : CGCTCAGT
    N726 : GTCTTAGG
    N727 : ACTGATCG
    N728 : TAGCTGCA
    N729 : GACGTCGA
</pre>

<br></br>

<h3>(1) Optional: stain cells/nulcei. Tagmentation on bulk cells/nulcei (will create 9-bp gap):</h3>
<img src="../data/tn5_dimer.svg" alt="Tn5 dimer" style="width:800px;height:450px;">
<pre>
<seq>
<i>Product 1 (s5 at both ends, not amplifiable due to semi-suppressiev PCR):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'


<i>Product 2 (s7 at both ends, not amplifiable due to semi-suppressiev PCR):</i>

5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                   <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 3 (different ends, amplifiable):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(2) After tagmentation nuclei are still intact, FACS sort them into wells containing lysis buffer, index primer, then add PCR reagent, 72 degree gap fill-in (the first cycle in Nextera PCR):</h3>
<pre>
<seq>
5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXXXXXXXXXXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>
    <s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXXXXXXXXXXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(7) Amplification using N/S5xx and N7xx index primers:</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[i5]<s5>TCGTCGGCAGCGTC</s5>---->
                                 5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXXXXXXXXXXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>
                                     <s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXXXXXXXXXXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
                                                                                                               <----<s7>GGCTCGGGTGCTCTG</s7>[i7]<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(8) Final library structure:</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXX...XXXXXXXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7>
    <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXXXXXXX...XXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
           <p5>Illumina P5</p5>              i5         <s5>s5</s5>              <me>ME</me>                cDNA                <me>ME</me>               <s7>s7</s7>          i7            <p7>Illumina P7</p7>
</align>
</pre>

<br></br>

<h2>Library sequencing:</h2>

<h3>(1) Add read 1 sequencing primer to sequence the first read (bottom strand as template):</h3>
<pre>
<align class="small">
                                     5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>------>
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXXXXXXX...XXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add index 1 sequencing primer to sequence the first index (i7) (bottom strand as template, 8 cycles):</h3>
<pre>
<align class="small">
                                                                                         5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>------>
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXXXXXXX...XXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Folds over and sequence the second index (i5) (bottom strand as template, 8 cycles. Single cells can be identified as the combination of i5 and i7):</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>------>
    <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXXXXXXX...XXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(4) Cluster regeneration, add read 2 sequencing primer to sequence the second read (top strand as template):</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXX...XXXXXXXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7>
                                                                                      <------<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</align>
</pre>

</body>
</html>
