<!DOCTYPE html>
<html>

<style>
seq {font-family:Monaco, "Courier New";
  font-size:0.9em;
  text-indent: 2.7em;
}

align.small {
    font-family:Monaco, "Courier New";
    font-size:0.9em;
    line-height:65%;
}

align.long {
  font-family:Monaco, "Courier New";
  font-size:0.7em;
  line-height:70%;
}

s5 {color:#6baed6;}
s7 {color:#fc9272;}
p5 {color:#08519c;}
p7 {color:#a50f15;}
me {color:#969696;}
tso {color:#2ca25f;}
cbc {color:#f768a1;}
umi {color:#807dba;}
t7 {color:blue;}
r1 {color:#4a1486;}
r2 {color:#6a51a3;}
r3 {color:#807dba;}
h3 {font-family:verdana;}

</style>

<head>
<title>SHARE-seq</title>
</head>
<body>

<h1><a href="https://www.sciencedirect.com/science/article/pii/S0092867420312538" target="_blank">SHARE-seq</a></h1>

<span style="font-family:verdana; font-size:1.1em;"><p>The SHARE-seq is developed based on the idea of combinatorial indexing stratgy that is used in <a href="https://teichlab.github.io/scg_lib_structs/methods_html/sci-RNA-seq.html" target="_blank">sci-RNA-seq</a> and <a href="https://teichlab.github.io/scg_lib_structs/methods_html/SPLiT-seq.html" target="_blank">SPLiT-seq</a>. The method introduced three rounds of barcodes by ligating barcoded adaptors to both RNA (gene expression) and tagmented DNA (chromatin accessibility) to achieve the multiomic profiling from the same single cells.</p></span>

<br></br>

<h2>Adapter and primer sequences:</h2>
<seq>
<p>RT_primer: 5'-/5Phos/ <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me><umi>[10-bp UMI]</umi>/iBiodT/TTTTTTTTTTTTTTVN -3'</p>
<p><sup>*</sup>Round1 barcodes: 5'-/5Phos/ <r1>CGCGCTGCATACTTG</r1><cbc>[8-bp Barcode1]</cbc><r1>CCCATGATCGTCCGA</r1> -3'</p>
<p>Round1 linker: 5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'</p>
<p><sup>*</sup>Round2 barcodes: 5'-/5Phos/<r2>CATCGGCGTACGACT</r2><cbc>[8-bp Barcode2]</cbc><r2>ATCCACGTGCTTGAG</r2> -3'</p>
<p>Round2 linker: 5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'</p>
<p><sup>*</sup>Round3 barcodes: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7><cbc>[8-bp Barcode3]</cbc><r3>GTGGCCGATGTTTCG</r3> -3'</p>
<p>Round3 linker: 5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'</p>
<p>Round1 blocking (reverse complementary to Round1 linker): 5'- <r1>CCCATGATCGTCCGA</r1><s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>Round2 blocking (reverse complementary to Round2 linker): 5'- <r2>ATCCACGTGCTTGAG</r2><r1>CGCGCTGCATACTTG</r1> -3'</p>
<p>Round3 blocking (reverse complementary to Round3 linker): 5'- <r3>GTGGCCGATGTTTCG</r3><r2>CATCGGCGTACGACT</r2> -3'</p>
<p>Template Switching Oligos (TSO): 5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATrGrG+G -3'</p>
<p>RNA_PCR_primer: 5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso> -3'</p>
<p>v2_Ad1.01: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[8-bp i5]<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTAT</me> -3'</p>
<p>Read 1 sequencing primer: 5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Index 1 sequencing primer: 5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7> -3'</p>
<p>Read 2 sequencing primer: 5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'
<p><sup>*</sup>Ther are 96 barcodes per round, to see the full sequence, check the <a href="../data/1-s2.0-S0092867420312538-mmc1.xlsx" target="_blank">Supplementary Table S1</a> from the SHARE-seq publication.</p>
</seq>


<br></br>

<h2>Step-by-step library generation</h2>
<h3>(1) Prepare ligation adapters by annealing barcodes with correponding linkers (in three different plates):</h3>
<pre>
<seq>
<i>Plate_R1: Round1 barcodes with Round1 linker:</i>

5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'
               3'- <r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1>/5Phos/ -5'


<i>Plate_R2: Round2 barcodes with Round2 linker:</i>

5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'
               3'- <r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2>/5Phos/ -5'


<i>Plate_R3: Round3 barcodes with Round3 linker:</i>

5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'
               3'- <r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'


</seq>
</pre>

<h3>(2) Perform ATAC (standard Illumina Tn5) to tag open chromatin DNA:</h3>
<img src="../data/tn5_dimer.svg" alt="Tn5 dimer" style="width:800px;height:450px;">
<pre>
<seq>
mRNA (unaffected, but cytoplasmic mRNA will probably leak out):

5'- XXXXXXXXXXXXXXXX...XXXXXXXXXXXXXXXX(pA) -3'

gDNA (three products):

<i>Product 1 (good one):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 2 (cannot be amplified):</i>

5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                   <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 3 (cannot be ligated):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'
</seq>
</pre>

<h3>(3) Perform reverse transcription <i>in situ</i> using the nuleus as the reaction chamber:</h3>
<pre>
<seq>
mRNA (unaffected, but cytoplasmic mRNA will probably leak out):

   <t7>1. First strand will be synthesised:</t7>

   5'- XXX...XXXB(pA) -3'
         <---XXNV(dT)/BiodT/<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'

   <t7>2. Then MMLV will add extra Cs allowing Template Switching Oligos (TSO) to be incorporated later on:</t7>

      5'-    XXX...XXXB(pA) -3'
      3'- CCCXXX...XXNV(dT)/BiodT/<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'

gDNA (unaffected in theory):

<i>Product 1 (good one):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 2 (cannot be amplified):</i>

5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                   <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 3 (cannot be ligated):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'
</seq>
</pre>

<h3>(4) Distritube to Plate_R1 for Round1 Ligation. Note the product 2 from gDNA in the previous step will not be able to amplify due to the PCR primer used in the final library PCR step. The product 3 from gDNA will not be able to ligate due to non compatible end with Plate_R1 overhang. Therefore, those two products from gDNA are omitted from here:</h3>
<pre>
<seq>
mRNA:

5'-    XXX...XXXB(pA) -3'                             5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'
3'- CCCXXX...XXNV(dT)/BiodT/<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1>/5Phos/ -5'


gDNA:

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1>/5Phos/ -5'
</seq>
</pre>

<h3>(5) Pool and split to Plate_R2 for Round2 Ligation:</h3>
<pre>
<align class="long">
mRNA:

5'-    XXX...XXXB(pA) -3'                             5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'
3'- CCCXXX...XXNV(dT)/BiodT/<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2>/5Phos/ -5'


gDNA:

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2>/5Phos/ -5'
</align>
</pre>

<h3>(6) Pool and split to Plate_R3 for Round3 Ligation:</h3>
<pre>
<align class="long">
mRNA:

5'-    XXX...XXXB(pA) -3'                             5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'       5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'
3'- CCCXXX...XXNV(dT)/BiodT/<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'


gDNA:

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'       5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(7) Pool, cell lysis, reverse crosslinking and physically separate gDNA and RNA using streptavidn pull down.</h3>

<h3>(8) Prepare ATAC library by amplify the gDNA using v2_Ad1.01 and Illumina P7 primers:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[8-bp i5]<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTAT</me>--------->
                                      5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'       5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'
                                                        <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
                                                                                                                                                                                                                                              <-----------------<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(9) For mRNA library preparation, it involves in the following steps:</h3>

<h3>(9.1) Template Switching Oligo (TSO) incorporation:</h3>
<pre>
<align class="long">
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXXB(pA) -3'                      5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'       5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXNV(dT)<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(9.2) cDNA amplification using RNA_PCR_primer and Illumina P7 primer:</h3>
<pre>
<align class="long">
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>------------>
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXXB(pA) -3'                      5'- <s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1> -3'       5'- <r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2> -3'       5'- <r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3> -3'
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXNV(dT)<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
                                                                                                                                                                                                   <-----------------<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(9.3) Purify the amplified cDNA:</h3>
<pre>
<align class="long">
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXXB(pA)<umi>[10-bp UMI]</umi><me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>[8-bp Barcode1]</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>[8-bp Barcode2]</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>[8-bp Barcode3]</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXNV(dT)<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(9.4) Tagmentation of the amplified cDNA using a Tn5 homodimer:</h3>
<img src="../data/s5_Tn5_dimer.svg" alt="Tn5 dimer" style="width:800px;height:450px;">
<pre>
<align class="long">
Prdocut 1 (left hand side of the fragment, not amplifiable):

5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'


Prdocut 2 (middle part of the fragment, not amplifiable):

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'


Product 3 (the right part of the fragment, the only amplifiable):

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXXB(pA)<umi>[10-bp UMI]</umi><me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>[8-bp Barcode1]</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>[8-bp Barcode2]</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>[8-bp Barcode3]</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXNV(dT)<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(9.5) Amplify mRNA library using v2_Ad1.01 and Illumina P7 primers:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[8-bp i5]<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTAT</me>--------->
                                      5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXXB(pA)<umi>[10-bp UMI]</umi><me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>[8-bp Barcode1]</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>[8-bp Barcode2]</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>[8-bp Barcode3]</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                        <me>TCTACACATATTCTCTGTC</me>         XXX...XXNV(dT)<umi>[10-bp UMI]</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>[8-bp Barcode1]</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>[8-bp Barcode2]</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>[8-bp Barcode3]</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
                                                                                                                                                                                                                                                     <-----------------<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(10) Final library structure:</h3>
<pre>
<seq>
mRNA:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXX...XXXB(pA)<umi>NNNNNNNNNN</umi><me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>NNNNNNNN</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>NNNNNNNN</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>NNNNNNNN</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXXV(dT)<umi>NNNNNNNNNN</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
            <p5>Illumina P5</p5>          8-bp i5       <s5>s5</s5>               <me>ME</me>           cDNA        <umi>10-bp UMI</umi>         <me>ME</me>               <s7>s7</s7>          <r1>linker1</r1>    <cbc>Barcode1</cbc>           <r2>linker2</r2>            <cbc>Barcode2</cbc>             <r3>linker3</r3>          <cbc>Barcode3</cbc>        <p7>Illumina P7</p7>


ATAC:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>NNNNNNNN</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>NNNNNNNN</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>NNNNNNNN</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
            <p5>Illumina P5</p5>          8-bp i5       <s5>s5</s5>               <me>ME</me>           gDNA          <me>ME</me>               <s7>s7</s7>          <r1>linker1</r1>     <cbc>Barcode1</cbc>           <r2>linker2</r2>            <cbc>Barcode2</cbc>             <r3>linker3</r3>          <cbc>Barcode3</cbc>        <p7>Illumina P7</p7>
</seq>
</pre>

<br></br>

<h2>Library sequencing:</h2>

<h3>(1) Add read 1 sequencing primer to sequence the first read (bottom strand as template, 30 cycles):</h3>
<pre>
<seq>
mRNA:
                                     5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>--->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXXV(dT)<umi>NNNNNNNNNN</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'


ATAC:
                                     5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>--->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</seq>
</pre>

<h3>(2) Add Index 1 sequencing primer to sequence cell barcodes (bottom strand as template, 99 cycles):</h3>
<pre>
<seq>
mRNA:
                                                                                              5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>-------------------------------------------------------------------------------------------------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXXV(dT)<umi>NNNNNNNNNN</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'


ATAC:
                                                                               5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>-------------------------------------------------------------------------------------------------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'

</seq>
</pre>

<h3>(3) Folds over and sequence the i5 index (bottom strand as template, s8 cycles):</h3>
<pre>
<seq>
mRNA:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>------>
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXXV(dT)<umi>NNNNNNNNNN</umi><me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'


ATAC:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>------>
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7><r1>AGCCTGCTAGTACCC</r1><cbc>NNNNNNNN</cbc><r1>GTTCATACGTCGCGC</r1><r2>GAGTTCGTGCACCTA</r2><cbc>NNNNNNNN</cbc><r2>TCAGCATGCGGCTAC</r2><r3>GCTTTGTAGCCGGTG</r3><cbc>NNNNNNNN</cbc><p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</seq>
</pre>

<h3>(4) Cluster regeneration, add Read 2 sequencing primer to sequence the UMI or gDNA (top strand as template 30 cycles):</h3>
<pre>
<seq>
mRNA:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXX...XXXB(pA)<umi>NNNNNNNNNN</umi><me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>NNNNNNNN</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>NNNNNNNN</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>NNNNNNNN</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                                 <----------------<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


ATAC:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7><r1>TCGGACGATCATGGG</r1><cbc>NNNNNNNN</cbc><r1>CAAGTATGCAGCGCG</r1><r2>CTCAAGCACGTGGAT</r2><cbc>NNNNNNNN</cbc><r2>AGTCGTACGCCGATG</r2><r3>CGAAACATCGGCCAC</r3><cbc>NNNNNNNN</cbc><p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                            <------<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'

</seq>
</pre>

<br></br>

</body>
</html>
